src/Chess.cpp

```diff
@@ -0,0 +1,257 @@
+			m_board[i][1] = m_board[i][19] = ('A' + t);
+	}
+}
+// put the pieces from the boardString 
+void Chess::setPieces()
+{
+	for (size_t row = 0, t = 0; row < 8; ++row)
+		for (size_t col = 0; col < 8; ++col, ++t)
+			m_board[(3 + (row * 2))][(3 + (col * 2))] = ((m_boardString[t] == '#') ? 32 : m_boardString[t]);
+}
+// print the only the board to screen 
+void Chess::show() const 
+{
+	for (size_t row = 0; row < _SIZE; ++row)
+	{
+		for (size_t col = 0; col < _SIZE; ++col)
+			cout << m_board[row][col];
+		cout << endl;
+	}
+}
+// clear screen and print the board and the relevant msg 
+void Chess::displayBoard() const
+{
+	clear();
+	show();
+	cout << m_msg<< m_errorMsg;
+	
+}
+// print the who is turn before getting input 
+void Chess::showAskInput() const 
+{
+	if (m_turn)
+		cout << "Player 1 (White - Capital letters) >> ";
+	else
+		cout << "Player 2 (Black - Small letters)   >> ";
+}
+// check if the source and dest are the same 
+bool Chess::isSame() const 
+{
+	return ((m_input[0] == m_input[2]) && (m_input[1] == m_input[3]));
+} 
+// check if the input is lockations at board
+bool Chess::isValid() const
+{
+	return ((('A' <= m_input[0]) && (m_input[0] <= 'H')) || (('a' <= m_input[0]) && (m_input[0] <= 'h')) &&
+		(('1' <= m_input[1]) && (m_input[1] <= '8')) &&
+		(('A' <= m_input[2]) && (m_input[2] <= 'H')) || (('a' <= m_input[2]) && (m_input[2] <= 'h')) &&
+		(('1' <= m_input[3]) && (m_input[3] <= '8')));
+}
+	
+// check if the input is exit or quit  
+bool Chess::isExit() const 
+{
+	return ((m_input == "exit") || (m_input == "quit") || (m_input == "EXIT") || (m_input == "QUIT"));
+}
+// execute the movement on board 
+void Chess::excute()
+{
+	int row = (m_input[0] - 'a');
+	int col = (m_input[1] - '1');
+	char pieceInSource = m_boardString[(row * 8) + col]; 
+	m_boardString[(row * 8) + col] = '#'; 
+
+	row = (m_input[2] - 'a');
+	col = (m_input[3] - '1');
+	m_boardString[(row * 8) + col] = pieceInSource; 
+
+	setPieces(); 
+}
+// check the response code and switch turn if needed 
+void Chess::doTurn()
+{
+	m_errorMsg = "\n"; 
+	switch (m_codeResponse)
+	{
+	case 11:
+	{
+		m_msg = "there is not piece at the source \n";
+		break;
+	}
+	case 12:
+	{
+		m_msg = "the piece in the source is piece of your opponent \n";
+		break;
+	}
+	case 13:
+	{
+		m_msg = "there one of your pieces at the destination \n";
+		break;
+	}
+	case 21:
+	{
+		m_msg = "illegal movement of that piece \n";
+		break;
+	}
+	case 31:
+	{
+		m_msg = "this movement will cause you checkmate \n";
+		break;
+	}
+	case 41:
+	{
+		excute();
+		m_turn = !m_turn;
+		m_msg = "the last movement was legal and cause check \n";
+		break;
+	}
+	case 42:
+	{
+		excute();
+		m_turn = !m_turn;
+		m_msg = "the last movement was legal \n";
+		break;
+	}
+	}
+}
+
+// C'tor
+Chess::Chess(const string& start)
+	: m_boardString(start),m_codeResponse(-1)
+{
+	setFrames();
+	setPieces();
+}
+
+// get the source and destination 
+string Chess::getInput()

```

Comment:
can be simplified and made more readable by breaking it down into smaller functions with clear responsibilities. This can improve code maintainability and readability.